pkg/loop/internal/test/config.go

```diff
@@ -85,3 +89,45 @@ func (s staticContractTransmitter) LatestConfigDigestAndEpoch(ctx context.Contex
 func (s staticContractTransmitter) FromAccount() (libocr.Account, error) {
 	return account, nil
 }
+
+type staticChainReader struct{}
+
+func (c staticChainReader) GetLatestValue(ctx context.Context, bc types.BoundContract, method string, params, retVal any) error {
+	if !assert.ObjectsAreEqual(bc, boundContract) {
+		return fmt.Errorf("expected report context %v but got %v", boundContract, bc)
+	}
+	if method != medianContractGenericMethod {
+		return fmt.Errorf("expected generic contract method %v but got %v", medianContractGenericMethod, method)
+	}
+
+	jsonParams, err := json.Marshal(params)
+	if err != nil {
+		return fmt.Errorf("Received non json-marshable params in GetLatestValue: %v", params)
+	}
+	var gotParams GetLatestValueParams
+	err = json.Unmarshal(jsonParams, &gotParams)
+	if err != nil {
+		return fmt.Errorf("Invalid params received in GetLatestValue, must be unmarshable into type %T", reflect.TypeOf(getLatestValueParams))
+	}
+
+	if !assert.ObjectsAreEqual(gotParams, getLatestValueParams) {
+		return fmt.Errorf("expected params %v but got %v", getLatestValueParams, gotParams)
+	}
+
+	jsonRet, err := json.Marshal(latestTransmissionDetails)
+	if err != nil {
+		return err
+	}
+
+	ret, ok := retVal.(*map[string]any)
+	if !ok {
+		return fmt.Errorf("Wrong type passed for retVal param to GetLatestValue impl (expected %T instead of %T", reflect.TypeOf(retVal), reflect.TypeOf(ret))
+	}
+
+	err = json.Unmarshal(jsonRet, ret)
+	if err != nil {
+		return err
+	}
```

Comment:
Is this meant to compare as well?

Reply:
The Unmarshal here is just returning a value in retVal... caller will compare it to the value it expects to receive back.  The type of the retVal is checked to make sure it's `map[string]any`, but as I understand it the type information is the only thing that's an input, the actual value is an output and only written to not read.